providers/structured/provider.go

```diff
@@ -28,6 +29,8 @@ type RecordEventCallArgs struct {
 }
 
 type StructuredOutputLogProvider struct {
+	providers.LogProvider
```

Comment:
This is actually the first time I've looked at the way we do log chaining, so let me know if this is incorrect, but does the next provider have to be explicitly called in all the various log methods below? (Not seeing it being invoked elsewhere.) I.e., in `func (p *StructuredOutputLogProvider) Error(...)`, you'd call 

    if p.nextProvider != nil {
        p.nextProvider.Error(ctx, report, args...)
    }

And same for the other six methods defined in `LogProvider`.

Reply:
I think you're right. Otherwise, it wouldn't really be chaining I guess.

Reply:
Sounds good- feel free to check that with Chris if there's any ambiguity, since there could be something I'm missing.